### DISCUSSION TITLE
[Feature Request] Add breakpoint support for Spacer

### DISCUSSION THREAD

**Discussion by 0xd8d** [heroui-inc/heroui#4680] [Feature requests]
### Is your feature request related to a problem? Please describe.

The current Spacer component only accepts an x and y prop to define the margins for their respective directions. In many cases, you'd want to have separate margins for mobile and desktop view. This is currently only possible by using multiple spacers and toggling their visibilities, which can quickly clutter the code.

### Describe the solution you'd like

The most intuitive use of such a component would be with additional props, such as the example below:
```jsx
<Spacer x={4} y={4} smX={6} smY={6} mdX={8} mdY={8} />
```

### Describe alternatives you've considered

In my current code base, we have created our own not so ideal hacky solution. This one only supports md.

```jsx
export const BRSpacer = ({ y, x, mdY, mdX }: Props) => {
  let keyForFirstSpacer = `spacer-${y}-${x}`;
  let BRSpace = [<Spacer key={keyForFirstSpacer} y={y} x={x} className={`${(mdY || mdX) && "md:hidden"}`} />];

  if (mdY || mdX) {
    let mdYValue = mdY ? mdY : y;
    let mdXValue = mdX ? mdX : x;
    let keyForSecondSpacer = `md-spacer-${mdY || y}-${mdX || x}`;
    BRSpace.push(<Spacer key={keyForSecondSpacer} y={mdYValue} x={mdXValue} className="hidden md:block" />);
  }

  return BRSpace;
};
```

**Comment by asyncmacro** [comment]
Tried working on it and got it similar to
```js
<Spacer x={4} y={4} smX={6} smY={6} lgX={8} lgY={8} />
```
Thus making it like `smX < x < lgX`. If this is a good idea I'd try making a PR
